ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -347,6 +358,24 @@ public void poll(long timeoutMs, List<NetworkSend> sends)
           close(key);
         }
       }
+
+      Iterator<Transmission> sslTransIter = pendingSslHandshakes.iterator();
+      while (sslTransIter.hasNext()) {
+        Transmission sslTransmission = sslTransIter.next();
+        if (sslTransmission != null && sslTransmission.ready()) {
+          connected.add(sslTransmission.getConnectionId());
+          this.metrics.selectorConnectionCreated.inc();
+          Long handshakeStartTime = sslHandshakeTimer.remove(sslTransmission.getConnectionId());
```

Comment:
I will update the patch with a minor fix. Someone can close it explicitly too by calling close(key)